### PR TITLE
Organization Form Redirect Issue Prevents User Navigation, Affecting User Experience.  Issue #808

### LINKED ISSUE #808: Critical Bug: Organization Form Redirect Issue Prevents User Navigation, Affecting User Experience
**Describe the bug**

**When a user clicks on the "Create Organization" button in the organization list page, the page switches to the organization creation form, but the URL remains the same. This makes it difficult for the user to go back to the organization list page or leave the organization form with any routing from the page. This could potentially confuse users who might want to go away from the organization form without submitting it.**

**To Reproduce**

- **Go to the organization list page (route: https://talawa.app/orglist)**
- **Click on the "Create Organization" button**
- **Notice that the page switches to the organization creation form, but the URL remains the same**
- **Try to navigate back to the organization list page or leave the organization form with any routing from the page**
- **Expected behavior**
- **The URL should change to reflect the organization creation form page, and the user should be able to navigate back to the organization list page or leave the organization form with any routing from the page.**

**Actual behavior**
**The URL remains the same, and the user cannot navigate back to the organization list page or leave the organization form with any routing from the page.**

**To Approach**
**One possible solution could be to implement a "Cancel" or "Go back" button on the organization form that takes the user back to the previous page or to a specific landing page. Another solution could be to change the URL when the organization form is displayed, perhaps by adding a query parameter or unique identifier to differentiate it from the previous page. This would allow users to easily navigate back to the previous page using the browser's back button or by changing the URL manually.**

**Potential internship candidates**
**Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359**

image

image

### PR DESCRIPTION
**Pull request by sumyat-aung**
**Description**

**This PR addresses issue #808 , which reported a bug where the organization creation form is displayed on the same URL as the organization list page when the "Create Organization" button is clicked, making it difficult for users to navigate back to the organization list page or leave the form with any routing from the page.**

**Approach**

**To solve this issue, I added a "Close" button to the organization creation modal, allowing users to easily close the modal and return to the organization list page. Additionally, I implemented a way to close the modal when the user clicks on the overlay, which is a common pattern in modal-based UIs.**

Fixes #808

<br />

image

<br />

**Changes Made**

- **Added a closing icon SVG in './src/assets/images/close'.**
- **Added that Closing button to the organization creation modal, which allows users to easily close the modal and return to the organization list page**
- **Implemented a way to close the modal when the user clicks on the overlay, which is a common pattern in modal-based UIs**

<br />

**Testing**

image

### CODE REVIEW

**Comment by palisadoes** [comment]
Your PR has been assigned reviewers. Please contact them either in this PR or via slack to get your code reviewed and approved. They should also be the:

1) arbitrators of future discussions with other contributors about the validity of your changes
2) point of contact for evaluating the validity of your work
3) person who verifies matching issues by others that should be closed.
4) person who assists in fixing your tests

Please make sure your code passes all tests. Our test code coverage system will fail if either of these two conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold

We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

**Comment by sumyat-aung** [comment]
@ttaylor92 
@Kevoniat

Would you mind taking a look at my PR when you have a chance? I've made the necessary changes for #808  and believe it's ready to be merged. Thank you!

**Review by Kevoniat — APPROVED** [review]

**Review by ttaylor92 — APPROVED** [review]

File: src/screens/OrgList/OrgList.tsx [hunk 1]
```diff
@@ -319,7 +322,13 @@ function OrgList(): JSX.Element {
                 className={styles.cancel}
                 data-testid="closeOrganizationModal"
               >
-                <i className="fa fa-times"></i>
+                <img
```
Comment (Rumone) [inline src/screens/OrgList/OrgList.tsx:325]: Why was the icon removed? The image would be an unnecessary overhead.

**Review by Rumone — CHANGES_REQUESTED** [review]

**Comment by palisadoes** [comment]
@sumyat-aung Please address the comment from @Rumone

File: src/screens/OrgList/OrgList.tsx (hunk 1 above)
Comment (sumyat-aung) [reply src/screens/OrgList/OrgList.tsx:325]: It seems that despite the code being present with `i` tag, the icon is not displaying on the page. I've already tried using 5 different icons from Font Awesome, but the issue still persists.

The image is an SVG, which is lightweight and shouldn't add unnecessary overhead to the page.

**Comment by sumyat-aung** [comment]
> @sumyat-aung Please address the comment from @Rumone

I did. 
When will I be able to merge? Both Reviewers has been approved.

File: src/screens/OrgList/OrgList.tsx (hunk 1 above)
Comment (palisadoes) [reply src/screens/OrgList/OrgList.tsx:325]: Each version of font-awesome has different names for its free icons. A free icon in v3 may not not be available in v4 and will therefore not display. Check the CSS for available icons to use and use the font-awesome website to see which ones are available for each version..

**Comment by sumyat-aung** [comment]
@Rumone 
@tonythegr8

Hello. I addressed the [requested changes](https://github.com/PalisadoesFoundation/talawa-admin/pull/815/commits/842a6e8b62ca367e99b51f147208cd792ca92e9a) by removing the image, and replacing it with correct version of font awesome. 
Please, review it when you have a chance? I believe it's ready to be merged. Thank you!

File: src/screens/OrgList/OrgList.tsx (hunk 1 above)
Comment (sumyat-aung) [reply src/screens/OrgList/OrgList.tsx]: @Rumone  
@palisadoes 

I addressed the request changes. Let me know if there's anything else to be done. Thanks

**Review by palisadoes — APPROVED** [review]

**Pull request merged**